lib/ProductOpener/Data.pm

```diff
@@ -119,6 +122,46 @@ sub execute_count_tags_query ($query) {
 	return execute_tags_query('count', $query);
 }
 
+sub execute_product_query ($parameters_ref, $query, $fields, $sort, $limit, $skip) {
```

Comment:
By convention we suffix references to object by _ref, so that would be $sort_ref and $query_ref

Reply:
and $fields_ref